Mutation in parameterized value is seen across repeated calls of the test-case

(couldn't find any relevant issue but #237)
I'm running a parameterized test-case with a mutable parameter `param_dict` that gets modified in the test-case code
(the param has a single value, for simplicity).
The test-case runs twice due to some other parameter, `run_twice`.
The modification leaks to the next test-case run;  
is that on purpose?

```python
import pytest

@pytest.fixture(params=[0, 1])
def run_twice(request):
    pass


@pytest.mark.parametrize("param_dict", [[1]])  # this parameter leaks across runs
def test_mutate_params(param_dict, run_twice):
    assert param_dict == [1]
    # mutation seen from subsequent test-case run.
    del param_dict[0]
```
...which passes only the 1st time:
```
platform linux -- Python 3.8.3, pytest-5.3.5, py-1.9.0, pluggy-0.13.1
rootdir: ../pytest-mutate_params
plugins: cov-2.8.1, xonsh-0.9.18
collected 2 items                                                                                                                                             

test_mutate_param.py .F                                                                                                                                 [100%]

================================== FAILURES ==================================
______________________ test_mutate_params[1-param_dict0] _____________________

param_dict = [], fixt = None

    @pytest.mark.parametrize("param_dict", [[1]])
    def test_mutate_params(param_dict, run_twice):
>       assert param_dict == [1]
E       assert [] == [1]
E         Right contains one more item: 1
E         Use -v to get the full diff

test_mutate_param.py:12: AssertionError
========================= 1 failed, 1 passed in 0.03s =========================

``` 

## Environment
```
$ pip list | grep pytest
pytest                             5.3.5
pytest-cov                         2.8.1
```
Linux, but i guess it doesn't make a difference.

--- a/doc/en/parametrize.rst
+++ b/doc/en/parametrize.rst
@@ -83,7 +83,7 @@ them in turn:
 
     Parameter values are passed as-is to tests (no copy whatsoever).
 
-    For example, if you pass a list or a dict as a parameter value, and 
+    For example, if you pass a list or a dict as a parameter value, and
     the test case code mutates it, the mutations will be reflected in subsequent
     test case calls.
 

--- a/doc/en/parametrize.rst
+++ b/doc/en/parametrize.rst
@@ -81,7 +81,8 @@ them in turn:
 
 .. note::
 
-    Parameter values are passed as-is to tests (no copy whatsoever), so if you mutate them,
+    Parameter values are passed as-is to tests (no copy whatsoever).
+    For example, if you pass a list or a dict as a parameter value, and mutate it,
     the mutations will be reflected in subsequent test case calls.
 
 .. note::

--- a/doc/en/parametrize.rst
+++ b/doc/en/parametrize.rst
@@ -81,23 +81,25 @@ them in turn:
 
 .. note::
 
-    1. Parameter values are passed as-is to tests (no copy whatsoever), so if you mutate them,
-       the mutations will be reflected in subsequent test case calls.
+    Parameter values are passed as-is to tests (no copy whatsoever), so if you mutate them,
+    the mutations will be reflected in subsequent test case calls.
 
-    2. pytest by default escapes any non-ascii characters used in unicode strings
-       for the parametrization because it has several downsides.
-       If however you would like to use unicode strings in parametrization
-       and see them in the terminal as is (non-escaped), use this option
-       in your ``pytest.ini``:
+.. note::
 
-       .. code-block:: ini
+    pytest by default escapes any non-ascii characters used in unicode strings
+    for the parametrization because it has several downsides.
+    If however you would like to use unicode strings in parametrization
+    and see them in the terminal as is (non-escaped), use this option
+    in your ``pytest.ini``:
 
-           [pytest]
-           disable_test_id_escaping_and_forfeit_all_rights_to_community_support = True
+    .. code-block:: ini
 
-       Keep in mind however that this might cause unwanted side effects and
-       even bugs depending on the OS used and plugins currently installed,
-       so use it at your own risk.
+        [pytest]
+        disable_test_id_escaping_and_forfeit_all_rights_to_community_support = True
+
+    Keep in mind however that this might cause unwanted side effects and
+    even bugs depending on the OS used and plugins currently installed,
+    so use it at your own risk.
 
 
 As designed in this example, only one pair of input/output values fails

--- a/doc/en/parametrize.rst
+++ b/doc/en/parametrize.rst
@@ -82,8 +82,10 @@ them in turn:
 .. note::
 
     Parameter values are passed as-is to tests (no copy whatsoever).
-    For example, if you pass a list or a dict as a parameter value, and mutate it,
-    the mutations will be reflected in subsequent test case calls.
+
+    For example, if you pass a list or a dict as a parameter value, and 
+    the test case code mutates it, the mutations will be reflected in subsequent
+    test case calls.
 
 .. note::
 

--- a/doc/en/parametrize.rst
+++ b/doc/en/parametrize.rst
@@ -81,17 +81,26 @@ them in turn:
 
 .. note::
 
-    pytest by default escapes any non-ascii characters used in unicode strings
-    for the parametrization because it has several downsides.
-    If however you would like to use unicode strings in parametrization and see them in the terminal as is (non-escaped), use this option in your ``pytest.ini``:
-
-    .. code-block:: ini
-
-        [pytest]
-        disable_test_id_escaping_and_forfeit_all_rights_to_community_support = True
-
-    Keep in mind however that this might cause unwanted side effects and
-    even bugs depending on the OS used and plugins currently installed, so use it at your own risk.
+    1. Paramater values are passed as-is to tests, and if you mutate them,
+       the mutations will be reflected in subsequent test case calls.
+   
+       Specifically, pytest doesn't do any copying of the values, since that would
+       put constraints on the possible parametrizable values, cause unexpected behaviour
+       due to broken copy/deepcopy, degrade performance, etc.
+   
+    2. pytest by default escapes any non-ascii characters used in unicode strings
+       for the parametrization because it has several downsides.
+       If however you would like to use unicode strings in parametrization
+       and see them in the terminal as is (non-escaped), use this option
+       in your ``pytest.ini``:
+   
+       .. code-block:: ini
+   
+           [pytest]
+           disable_test_id_escaping_and_forfeit_all_rights_to_community_support = True
+   
+       Keep in mind however that this might cause unwanted side effects and
+       even bugs depending on the OS used and plugins currently installed, so use it at your own risk.
 
 
 As designed in this example, only one pair of input/output values fails

--- a/doc/en/parametrize.rst
+++ b/doc/en/parametrize.rst
@@ -81,26 +81,27 @@ them in turn:
 
 .. note::
 
-    1. Paramater values are passed as-is to tests, and if you mutate them,
+    1. Parameter values are passed as-is to tests, and if you mutate them,
        the mutations will be reflected in subsequent test case calls.
-   
+
        Specifically, pytest doesn't do any copying of the values, since that would
        put constraints on the possible parametrizable values, cause unexpected behaviour
        due to broken copy/deepcopy, degrade performance, etc.
-   
+
     2. pytest by default escapes any non-ascii characters used in unicode strings
        for the parametrization because it has several downsides.
        If however you would like to use unicode strings in parametrization
        and see them in the terminal as is (non-escaped), use this option
        in your ``pytest.ini``:
-   
+
        .. code-block:: ini
-   
+
            [pytest]
            disable_test_id_escaping_and_forfeit_all_rights_to_community_support = True
-   
+
        Keep in mind however that this might cause unwanted side effects and
-       even bugs depending on the OS used and plugins currently installed, so use it at your own risk.
+       even bugs depending on the OS used and plugins currently installed,
+       so use it at your own risk.
 
 
 As designed in this example, only one pair of input/output values fails

--- a/doc/en/parametrize.rst
+++ b/doc/en/parametrize.rst
@@ -81,12 +81,8 @@ them in turn:
 
 .. note::
 
-    1. Parameter values are passed as-is to tests, and if you mutate them,
+    1. Parameter values are passed as-is to tests (no copy whatsoever), so if you mutate them,
        the mutations will be reflected in subsequent test case calls.
-
-       Specifically, pytest doesn't do any copying of the values, since that would
-       put constraints on the possible parametrizable values, cause unexpected behaviour
-       due to broken copy/deepcopy, degrade performance, etc.
 
     2. pytest by default escapes any non-ascii characters used in unicode strings
        for the parametrization because it has several downsides.

